web/src/app/webview/src/contextManager.ts

```diff
@@ -197,7 +197,7 @@ export default class ContextManager extends ContextManagerBase<WebviewConfigurat
     return activatingKeyboard;
   }
 
-  public resetContext(): void {
+  public async resetContext(): Promise<void> {
     super.resetContext();
```

Comment:
Or do I have to await that as well?
```suggestion
    await super.resetContext();
```

Reply:
Yes, without the `await` the Promise is not returned to the caller, which makes the `async` definition somewhat pointless. There are many potential promise races in the KeymanWeb codebase, and really almost all of the Promises need to be explicitly returned to the caller (and `async`/`await` makes this simpler) -- and if a promise is _not_ returned to the caller, this needs to be very clearly documented along with a rationale. As it stands, it's very hard indeed to trace all the code paths and places where promises may be leaking out.

See also https://github.com/keymanapp/keyman/pull/11424#discussion_r1610996111 which looks like it is touching on the same thing so you may need to coordinate with @jahorton.

Reply:
Thanks. Moved this commit as a separate PR #11605.